processor/routingprocessor/router.go

```diff
@@ -95,10 +95,10 @@ func (r *router) routeMetricsForResource(_ context.Context, tm pdata.Metrics) []
 		}
 
 		if rEntry, ok := routingMap[attrValue]; ok {
-			resMetrics.CopyTo(rEntry.resMetrics.AppendEmpty())
```

Comment:
Looks good, I have a similar change as part of #7771 but I think @dmitryax had some concerns. What do you think about using MoveTo here, @dmitryax?

Reply:
No, this is different. I believe we can switch to `MoveTo` here, because it's called only once per each `ResourceMetrics` object, and we don't need to preserve the original `pdata.Metrics` object. But it's not the case for #7771.